zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -304,8 +295,18 @@ where
             reconnect
         };
 
-        // SECURITY: rate-limit new candidate connections
-        sleep.await;
+        // # Security
+        //
+        // Rate-limit new candidate connections to avoid denial of service.
+        //
+        // Update the deadline before sleeping, so we handle concurrent requests
+        // correctly.
+        let current_deadline = self.next_peer_sleep_until;
+        // If we recently had a connection, and we're about to sleep, base the
+        // next time off our sleep time. Otherwise, use the current time.
+        self.next_peer_sleep_until = max(self.next_peer_sleep_until, Instant::now());
+        self.next_peer_sleep_until += constants::MIN_PEER_CONNECTION_INTERVAL;
+        sleep_until(current_deadline).await;
```

Comment:
Security: Correctly use MIN_PEER_CONNECTION_INTERVAL in CandidateSet

- Make sure the next deadline is at least `MIN_PEER_CONNECTION_INTERVAL` in the future
- Replace the `Sleep` future with an `Instant`